dev/tools/update-compat.py

```diff
@@ -0,0 +1,254 @@
+#!/usr/bin/env python
+from __future__ import with_statement
+import os, re, sys
+
+SCRIPT_PATH = os.path.dirname(os.path.realpath(__file__))
+ROOT_PATH = os.path.realpath(os.path.join(SCRIPT_PATH, '..', '..'))
+CONFIGURE_PATH = os.path.join(ROOT_PATH, 'configure.ml')
+HEADER_PATH = os.path.join(ROOT_PATH, 'dev', 'header.ml')
+NUMBER_OF_OLD_VERSIONS = 2
+NUMBER_OF_COMPAT_VERSIONS = NUMBER_OF_OLD_VERSIONS+1
+EXTRA_HEADER = '\n(** Compatibility file for making Coq act similar to Coq v%s *)\n'
+FLAGS_MLI_PATH = os.path.join(ROOT_PATH, 'lib', 'flags.mli')
+FLAGS_ML_PATH = os.path.join(ROOT_PATH, 'lib', 'flags.ml')
+COQARGS_ML_PATH = os.path.join(ROOT_PATH, 'toplevel', 'coqargs.ml')
+G_VERNAC_PATH = os.path.join(ROOT_PATH, 'vernac', 'g_vernac.mlg')
+TEST_SUITE_PATHS = tuple(os.path.join(ROOT_PATH, 'test-suite', 'success', i)
+                         for i in ('CompatOldFlag.v', 'CompatPreviousFlag.v', 'CompatCurrentFlag.v'))
+TEST_SUITE_DESCRIPTIONS = ('current-minus-two', 'current-minus-one', 'current')
+# sanity check that we are where we think we are
+assert(os.path.normpath(os.path.realpath(SCRIPT_PATH)) == os.path.normpath(os.path.realpath(os.path.join(ROOT_PATH, 'dev', 'tools'))))
```

Comment:
Does this mean that the script has to be run from inside the `dev/tools` directory? Some of the scripts in there assume on the contrary that they are run as `dev/tools/script` from the root of the repo.

Reply:
No.  It assumes that the script lives in `dev/tools`, though; copying it somewhere else will not work.  I define `SCRIPT_PATH` based on `__file__` so that it does not matter what the current directory is.  Perhaps this deserves a comment?

Reply:
More comments is always good. But OK, this is great and robust :tada:

Reply:
Updated with comment